src/PermissionedUSDCWrapper.sol

```diff
@@ -0,0 +1,209 @@
+        }
+        _burn(_msgSender(), value);
+        SafeERC20.safeTransfer(_underlying, account, value);
+        return true;
+    }
+
+    function transferFrom(address from, address to, uint256 value) public override returns (bool) {
+        require(hasPermission(to), "USDCWrapper: no permission");
+        address spender = _msgSender();
+        _spendAllowance(from, spender, value);
+        _transfer(from, to, value);
+        return true;
+    }
+
+    function transfer(address to, uint256 value) public override returns (bool) {
+        require(hasPermission(to), "USDCWrapper: no permission");
+        address owner = _msgSender();
+        _transfer(owner, to, value);
+        return true;
+    }
+
+    function hasPermission(address account) public view returns (bool attested) {
+        if (account == address(this) || account == address(0) || account == MORPHO || account == BUNDLER || memberlist.isMember(account)) {
+            return true;
+        }
+
+        Attestation memory verifiedAccountAttestation = getVerifiedAccountAttestation(account);
+        bool isAccountVerified = keccak256(verifiedAccountAttestation.data) == keccak256(abi.encodePacked(uint256(1)));
+        
+        Attestation memory verifiedCountryAttestation = getVerifiedCountryAttestation(account);
+        string memory countryCode = parseCountryCode(verifiedCountryAttestation.data);
+        bool isUS = keccak256(abi.encodePacked(countryCode)) == keccak256(abi.encodePacked("US"));
+
+        return isAccountVerified && !isUS;
+    }
+
+    function getVerifiedCountryAttestation(address account) public view returns (Attestation memory attestation) {
+        bytes32 attestationUid = attestationIndexer.getAttestationUid(account, verifiedCountrySchemaUid);
+        require(attestationUid != 0, "USDCWrapper: no attestation found");
+        attestation = attestationService.getAttestation(attestationUid);
+        require(attestation.expirationTime == 0, "USDCWrapper: attestation expired");
+        require(attestation.revocationTime == 0, "USDCWrapper: attestation revoked");
+    }
+
+    function getVerifiedAccountAttestation(address account) public view returns (Attestation memory attestation) {
+        bytes32 attestationUid = attestationIndexer.getAttestationUid(account, verifiedAccountSchemaUid);
+        require(attestationUid != 0, "USDCWrapper: no attestation found");
+        attestation = attestationService.getAttestation(attestationUid);
+        require(attestation.expirationTime == 0, "USDCWrapper: attestation expired");
+        require(attestation.revocationTime == 0, "USDCWrapper: attestation revoked");
+    }
+
+    /**
+     * @dev Mint wrapped token to cover any underlyingTokens that would have been transferred by mistake or acquired from
+     * rebasing mechanisms. Internal function that can be exposed with access control if desired.
+     */
+    function _recover(address account) internal override returns (uint256) {
```

Comment:
As an internal method, it is useless. I would actually expose this, with `auth`.